### ISSUE TITLE
'zip' is not recognized as an internal or external command, operable program or batch file.

### ISSUE THREAD

**Issue by kmgalanakis** [wp-cli/dist-archive-command#22]
I'm on windows and even though I can run the `zip` command on my terminal, I keep getting this error

`'zip' is not recognized as an internal or external command, operable program or batch file.`

Any clues why this happens?

**Comment by gitlost** [comment]
Yes I do @kmgalanakis ! The framework needs changes to properly support Windows. In this case the issue is caused (ultimately) by the environment (in particular `PATH`) not being passed to `proc_open()`. Also this command should probably use the `Extractor` class anyway instead.

**Comment by kmgalanakis** [comment]
Minutes after posting the issue, I dug through the code and immediately noticed the issue the `proc_open`. Is there any roadmap for that refactor?

**Comment by gitlost** [comment]
Always the way isn't it?! The roadmap is here https://make.wordpress.org/cli/handbook/roadmap/ and as you can see is currently a bit vague but will hopefully be firmed up over the coming week or so...

**Comment by bondansebastian** [comment]
Experiencing the same issue, any updates ?

**Comment by wojsmol** [comment]
Hi
@bondansebastian This commend is reedy yet for WP-CLI 2.. If you wont help as in converting it to new format see https://github.com/wp-cli/dist-archive-command/issues/26 and join as on slack in #cli channal.

**Comment by wojsmol** [comment]
@bondansebastian To be clear this command will by converted shortly.

**Comment by bondansebastian** [comment]
Hi @wojsmol 
[#26](https://github.com/wp-cli/dist-archive-command/issues/26) has been closed, still the command doesn't work in windows machine.
I would be glad to help converting it, just need more information about what to fix

**Comment by wojsmol** [comment]
@bondansebastian Command was converted to WP-CLI 2 structure. Please post `wp --info` output.

**Comment by bondansebastian** [comment]
> @bondansebastian Command was converted to WP-CLI 2 structure. Please post `wp --info` output.

    OS:     Windows NT 10.0 build 17134 (Windows 10) i586
    Shell:  C:\Program Files\Git\usr\bin\bash.exe
    PHP binary:     C:/xampp/php/php
    PHP version:    7.1.15
    php.ini used:   C:\xampp\php\php.ini
    WP-CLI root dir:        C:\Users\Bondan\AppData\Roaming\Composer\vendor\wp-cli\wp-cli
    WP-CLI vendor dir:      C:\Users\Bondan\AppData\Roaming\Composer\vendor
    WP_CLI phar path:
    WP-CLI packages dir:    C:\Users\Bondan/.wp-cli/packages/
    WP-CLI global config:
    WP-CLI project config:
    WP-CLI version: 2.1.0

**Comment by venezianluis** [comment]
Hi @kmgalanakis 
Here's my solution to zip files using Windows 10 & PowerShell Scripts: [InfoZip](http://infozip.sourceforge.net/)

- Find download section: [ftp://ftp.info-zip.org/pub/infozip/win32/]( ftp://ftp.info-zip.org/pub/infozip/win32/zip300xn-x64.zip)
- Find your file, there are many but i only needed this: [ftp://ftp.info-zip.org/pub/infozip/win32/zip300xn-x64.zip](ftp://ftp.info-zip.org/pub/infozip/win32/zip300xn-x64.zip)
- Put zip.exe path as a enviroment variable path and it's done!

After all you can zip everything by using `zip destination.zip source`
Try `zip --help` to see many others functions such as recursive zip into directories.

**Comment by kmgalanakis** [comment]
@venezianluis I've already tried this. Didn't resolve my issue.

**Comment by Lazar014** [comment]
Hey people, any solution to this problem yet?

**Comment by VoHoTv** [comment]
Is there any reason why the PHP ZipArchive is not used instead of relying on the system to have a `zip` command implemented?
